railties/lib/rails/backtrace_cleaner.rb

```diff
@@ -4,18 +4,26 @@
 
 module Rails
   class BacktraceCleaner < ActiveSupport::BacktraceCleaner
-    APP_DIRS_PATTERN = /^\/?(app|config|lib|test|\(\w*\))/
+    APP_DIRS_PATTERN = /\A\/?(?:app|config|lib|test|\(\w*\))/
     RENDER_TEMPLATE_PATTERN = /:in `.*_\w+_{2,3}\d+_\d+'/
-    EMPTY_STRING = ""
-    SLASH        = "/"
-    DOT_SLASH    = "./"
 
     def initialize
       super
       @root = "#{Rails.root}/"
-      add_filter { |line| line.sub(@root, EMPTY_STRING) }
-      add_filter { |line| line.sub(RENDER_TEMPLATE_PATTERN, EMPTY_STRING) }
-      add_filter { |line| line.sub(DOT_SLASH, SLASH) } # for tests
+      add_filter do |line|
+        line.start_with?(@root) ? line[@root.size, line.size] : line
```

Comment:
Would the `[]` calls be clearer if we used AS' `from`, aka `line.from(@root.size)`?

Reply:
👍 Much cleaner!